crates/router/src/core/payments.rs

```diff
@@ -193,7 +193,11 @@ where
                 .await?
             }
         };
-        vault::Vault::delete_locker_payment_method_by_lookup_key(state, &payment_data.token).await
+        if payment_data.payment_intent.status != storage_enums::IntentStatus::RequiresCustomerAction
```

Comment:
Why this change is added?

Reply:
We need to keep the token till will get the terminal status, RequiresCustomerAction is not a terminal status so it should be deleted only after completing authorize step